validation/src/block_production.rs

```diff
@@ -175,6 +185,8 @@ impl<Client, TxPool, Backend> consensus::Proposer<Block> for Proposer<Client, Tx
 		let transaction_pool = self.transaction_pool.clone();
 		let table = self.tracker.table().clone();
 		let backend = self.backend.clone();
+		let metrics = self.metrics.clone();
+		let block_timer = metrics.report(|metrics| metrics.block_constructed.start_timer());
```

Comment:
This should be moved into the `async move`. AFAIK the async code gets called on the first poll?

Reply:
Yeah, but I doubt this will make much difference